internal/service/cloud/network.go

```diff
@@ -162,6 +164,11 @@ func (s *Service) getLAN(ctx context.Context, ms *scope.Machine) (*sdk.Lan, erro
 	)
 
 	for _, l := range *lans.Items {
+		// Use an existing LAN if it's specified in the spec.
+		if ms.IonosMachine.Spec.LanID != nil && cmp.Equal(l.GetId(), ms.IonosMachine.Spec.LanID) {
```

Comment:
upstream k8s.io/utils has [ptr.Equal](https://github.com/kubernetes/utils/blob/master/ptr/ptr.go#L65). We have local copy to prevent having k8s.io/utils as direct dependency. It think it would be the better choice here.

Reply:
will check it.